node/coinstacks/solana/api/src/app.ts

```diff
@@ -45,4 +58,30 @@ app.get('/', async (_, res) => {
 
 app.use(middleware.errorHandler, middleware.notFoundHandler)
 
-app.listen(PORT, () => logger.info('Server started'))
+const server = app.listen(PORT, () => logger.info('Server started'))
+
+const transactionHandler: TransactionHandler<GeyserResultTransaction, GeyserResultTransaction> = async (geyserTx) => {
```

Comment:
The main meat here will be needing to convert a websocket transaction into an unchained api solana `Tx`. The unchained-client parser will be expecting the same transaction format to parse, and we will want that to be the human readable "parsed transaction" view as returned from the enhanced api.

Reply:
I'm not sure to get it

This is already the case, we return the parsed transaction from the enhanced API

```json
{
  "transaction": {
    "signatures": [
      "zdvyeeAPwCnJjyBMK5aaBgjHo7SLZSpNHBAnbWWuYhEvkdx2ENAaG8sGHWS13vXE3zdbthbNgVVmXsxgAD5Y3ia"
    ],
    "message": {
      "accountKeys": [
        {
          "pubkey": "9cGjV3ZF6yKgtWJBCCWUvFeouxVSMTioW9Vtfjr7ZEKD",
          "writable": true,
          "signer": true,
          "source": "transaction"
        },
        {
          "pubkey": "2jRuZvxbbe3A3P1DPFvwH1pt2NXxteRTyqaVKbcby1Ds",
          "writable": true,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "FwYVAHL4dd5cGrMexZhe8frGGegokHinekCMdG9cFHab",
          "writable": true,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "83yLKHg3X9ahyKmaiZc3CCdo3cxaCyiywcNbRKdqP1sJ",
          "writable": true,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "4a6vX1nDn7PLMUTpxDCxRMN17y4Cu9fGu7agE3pVRpdh",
          "writable": true,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "7jWrQYjfuHyQXVWfMyLireeukSpva99FvCLLxERCvT4U",
          "writable": true,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "Gv4Z1UoYsEbjbocKDTD9P9CVBrBjUwJxq6h49xAdepmz",
          "writable": true,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "7Krk48yc6SVrzUMFM4zScJ9vjvdcLhtdCDAjf1RncxqM",
          "writable": true,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "3G2eKnZ4RCEsDSbNWLtXamqs2tgmF2BzAD4CCc4cYTwJ",
          "writable": true,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "AiQZmhgByVN36qFALH5M1tiv3eJ3cEXQtU7LcLxE1Piz",
          "writable": true,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "foodQJAztMzX1DKpLaiounNe2BDMds5RNuPC6jsNrDG",
          "writable": true,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "ammoK8AkX2wnebQb35cDAZtTkvsXQbi82cGeTnUvvfK",
          "writable": true,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "ComputeBudget111111111111111111111111111111",
          "writable": false,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "SAGE2HAwep459SNq61LHvjxPk4pLPEJLoMETef7f7EE",
          "writable": false,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "FpTUZKuviuGaww6ijjXdoeuJtFeEjabEXnzxRYHukhMx",
          "writable": false,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "94LdKdSHuG3Na6H1YhkgJsq1caYVxUNeBRm7rLB6hd8k",
          "writable": false,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "GYFk8mgxw2M8uWLrL1hCfYDMLfAZLDH2VEzcrBoJwgue",
          "writable": false,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "EhdwsXyEuqYdfxCaYcYMxehDt8PpcPJ3q54aeQEa9E1x",
          "writable": false,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "DpMGTLKi9ZYpLsoD1R3tBwxpnzirh1hsRgAbJYq6Nnou",
          "writable": false,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "CSTatsVpHbvZmwHbCjZKVfYQT5JXfsXccXufhEcwCqTg",
          "writable": false,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "FX6ETuCcYSTxoH7G77cJgKorXK2x2Khx9wuZUEdTXyfm",
          "writable": false,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "GAMEzqJehF8yAnKiTARUuhZMvLvkZVAsCVri5vSfemLr",
          "writable": false,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "Cargo2VNTPPTi9c1vq1Jw5d3BWUNr18MjRtSupAghKEk",
          "writable": false,
          "signer": false,
          "source": "transaction"
        },
        {
          "pubkey": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
          "writable": false,
          "signer": false,
          "source": "transaction"
        }
      ],
      "recentBlockhash": "8fb9LaStGvYsBqdLExQdHhJq45PKDeX6KpZJfViP3hRV",
      "instructions": [
        {
          "programId": "ComputeBudget111111111111111111111111111111",
          "accounts": [],
          "data": "3DdGGhkhJbjm",
          "stackHeight": null
        },
        {
          "programId": "SAGE2HAwep459SNq61LHvjxPk4pLPEJLoMETef7f7EE",
          "accounts": [
            "2jRuZvxbbe3A3P1DPFvwH1pt2NXxteRTyqaVKbcby1Ds",
            "FwYVAHL4dd5cGrMexZhe8frGGegokHinekCMdG9cFHab",
            "83yLKHg3X9ahyKmaiZc3CCdo3cxaCyiywcNbRKdqP1sJ",
            "FpTUZKuviuGaww6ijjXdoeuJtFeEjabEXnzxRYHukhMx",
            "4a6vX1nDn7PLMUTpxDCxRMN17y4Cu9fGu7agE3pVRpdh",
            "7jWrQYjfuHyQXVWfMyLireeukSpva99FvCLLxERCvT4U",
            "94LdKdSHuG3Na6H1YhkgJsq1caYVxUNeBRm7rLB6hd8k",
            "Gv4Z1UoYsEbjbocKDTD9P9CVBrBjUwJxq6h49xAdepmz",
            "7Krk48yc6SVrzUMFM4zScJ9vjvdcLhtdCDAjf1RncxqM",
            "3G2eKnZ4RCEsDSbNWLtXamqs2tgmF2BzAD4CCc4cYTwJ",
            "AiQZmhgByVN36qFALH5M1tiv3eJ3cEXQtU7LcLxE1Piz",
            "foodQJAztMzX1DKpLaiounNe2BDMds5RNuPC6jsNrDG",
            "ammoK8AkX2wnebQb35cDAZtTkvsXQbi82cGeTnUvvfK",
            "GYFk8mgxw2M8uWLrL1hCfYDMLfAZLDH2VEzcrBoJwgue",
            "EhdwsXyEuqYdfxCaYcYMxehDt8PpcPJ3q54aeQEa9E1x",
            "DpMGTLKi9ZYpLsoD1R3tBwxpnzirh1hsRgAbJYq6Nnou",
            "CSTatsVpHbvZmwHbCjZKVfYQT5JXfsXccXufhEcwCqTg",
            "FX6ETuCcYSTxoH7G77cJgKorXK2x2Khx9wuZUEdTXyfm",
            "GAMEzqJehF8yAnKiTARUuhZMvLvkZVAsCVri5vSfemLr",
            "Cargo2VNTPPTi9c1vq1Jw5d3BWUNr18MjRtSupAghKEk",
            "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA"
          ],
          "data": "L7u68T6jgbD",
          "stackHeight": null
        }
      ],
      "addressTableLookups": []
    }
  },
  "meta": {
    "err": null,
    "status": {
      "Ok": null
    },
    "fee": 5001,
    "preBalances": [
      102977237,
      4621440,
      1712160,
      1712160,
      1760880,
      2143680,
      2039280,
      2039280,
      2039280,
      2039280,
      1461600,
      1461600,
      1,
      1141440,
      1920960,
      3132000,
      1712160,
      1712160,
      1712160,
      1426800,
      16112400,
      7231440,
      1141440,
      934087680
    ],
    "postBalances": [
      102972236,
      4621440,
      1712160,
      1712160,
      1760880,
      2143680,
      2039280,
      2039280,
      2039280,
      2039280,
      1461600,
      1461600,
      1,
      1141440,
      1920960,
      3132000,
      1712160,
      1712160,
      1712160,
      1426800,
      16112400,
      7231440,
      1141440,
      934087680
    ],
    "innerInstructions": [],
    "logMessages": [
      "Program ComputeBudget111111111111111111111111111111 invoke [1]",
      "Program ComputeBudget111111111111111111111111111111 success",
      "Program SAGE2HAwep459SNq61LHvjxPk4pLPEJLoMETef7f7EE invoke [1]",
      "Program log: Instruction: FleetStateHandler",
      "Program log: Current state: Idle(Idle { sector: [40, 30] })",
      "Program SAGE2HAwep459SNq61LHvjxPk4pLPEJLoMETef7f7EE consumed 10299 of 199850 compute units",
      "Program SAGE2HAwep459SNq61LHvjxPk4pLPEJLoMETef7f7EE success"
    ],
    "preTokenBalances": [
      {
        "accountIndex": 6,
        "mint": "foodQJAztMzX1DKpLaiounNe2BDMds5RNuPC6jsNrDG",
        "uiTokenAmount": {
          "uiAmount": 0,
          "decimals": 0,
          "amount": "0",
          "uiAmountString": "0"
        },
        "owner": "FwYVAHL4dd5cGrMexZhe8frGGegokHinekCMdG9cFHab",
        "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA"
      },
      {
        "accountIndex": 7,
        "mint": "ammoK8AkX2wnebQb35cDAZtTkvsXQbi82cGeTnUvvfK",
        "uiTokenAmount": {
          "uiAmount": 26634,
          "decimals": 0,
          "amount": "26634",
          "uiAmountString": "26634"
        },
        "owner": "83yLKHg3X9ahyKmaiZc3CCdo3cxaCyiywcNbRKdqP1sJ",
        "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA"
      },
      {
        "accountIndex": 8,
        "mint": "HYDR4EPHJcDPcaLYUcNCtrXUdt1PnaN4MvE655pevBYp",
        "uiTokenAmount": {
          "uiAmount": 244971173803,
          "decimals": 0,
          "amount": "244971173803",
          "uiAmountString": "244971173803"
        },
        "owner": "FpTUZKuviuGaww6ijjXdoeuJtFeEjabEXnzxRYHukhMx",
        "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA"
      },
      {
        "accountIndex": 9,
        "mint": "HYDR4EPHJcDPcaLYUcNCtrXUdt1PnaN4MvE655pevBYp",
        "uiTokenAmount": {
          "uiAmount": 61589,
          "decimals": 0,
          "amount": "61589",
          "uiAmountString": "61589"
        },
        "owner": "FwYVAHL4dd5cGrMexZhe8frGGegokHinekCMdG9cFHab",
        "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA"
      }
    ],
    "postTokenBalances": [
      {
        "accountIndex": 6,
        "mint": "foodQJAztMzX1DKpLaiounNe2BDMds5RNuPC6jsNrDG",
        "uiTokenAmount": {
          "uiAmount": 0,
          "decimals": 0,
          "amount": "0",
          "uiAmountString": "0"
        },
        "owner": "FwYVAHL4dd5cGrMexZhe8frGGegokHinekCMdG9cFHab",
        "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA"
      },
      {
        "accountIndex": 7,
        "mint": "ammoK8AkX2wnebQb35cDAZtTkvsXQbi82cGeTnUvvfK",
        "uiTokenAmount": {
          "uiAmount": 26634,
          "decimals": 0,
          "amount": "26634",
          "uiAmountString": "26634"
        },
        "owner": "83yLKHg3X9ahyKmaiZc3CCdo3cxaCyiywcNbRKdqP1sJ",
        "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA"
      },
      {
        "accountIndex": 8,
        "mint": "HYDR4EPHJcDPcaLYUcNCtrXUdt1PnaN4MvE655pevBYp",
        "uiTokenAmount": {
          "uiAmount": 244971173803,
          "decimals": 0,
          "amount": "244971173803",
          "uiAmountString": "244971173803"
        },
        "owner": "FpTUZKuviuGaww6ijjXdoeuJtFeEjabEXnzxRYHukhMx",
        "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA"
      },
      {
        "accountIndex": 9,
        "mint": "HYDR4EPHJcDPcaLYUcNCtrXUdt1PnaN4MvE655pevBYp",
        "uiTokenAmount": {
          "uiAmount": 61589,
          "decimals": 0,
          "amount": "61589",
          "uiAmountString": "61589"
        },
        "owner": "FwYVAHL4dd5cGrMexZhe8frGGegokHinekCMdG9cFHab",
        "programId": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA"
      }
    ],
    "rewards": [],
    "computeUnitsConsumed": 10449
  },
  "version": 0
}
```

Reply:
That is the parsed rpc transaction format, not the enhanced (enriched) transaction format as returned by: https://docs.helius.dev/solana-apis/enhanced-transactions-api/parse-transaction-s. The enhanced tx structure is certainly more readable, but may be hard for us to parse in exactly the same manner (maybe). So we have 3 options:
- convert parsed rpc transaction -> helius enhanced transaction format if possible
- type an unchained solana tx structure that both the parsed rpc transaction and helius enhanced transaction can be converted to
- switch the websockets to just transaction signatures and make a request to the enhanced transaction api for every message at the cost of req consumption

Reply:
Yeah since we are already filtering on addresses via the websocket we are going to have a much reduced data set to manage. It will add to the latency of push updates, but not much we can do about that. I would just start with all txs, no status filter because we would want to show failed txs on the front end regardless.

Reply:
Then, if we go for all TXs, it means unchained should do that call already and return the enriched TX right?

I wanted to let the client do the enriched calls to reduce the calls initially

Reply:
Correct, we want unchained to return the same tx format from `getTxHistory` `getTransaction` and `subscribeAddresses`. The web client parser will be expecting the same format across the board for parsing the transactions accordingly and shouldn't be required to make an extra call to make that happen.